scripts/stubsabot.py

```diff
@@ -239,27 +241,27 @@ class GithubInfo:
     tags: list[dict[str, Any]]
 
 
-async def get_github_repo_info(session: aiohttp.ClientSession, pypi_info: PypiInfo) -> GithubInfo | None:
+async def get_github_repo_info(session: aiohttp.ClientSession, stub_info: StubInfo) -> GithubInfo | None:
     """
-    If the project represented by `pypi_info` is hosted on GitHub,
+    If the project represented by `stub_info` is hosted on GitHub,
     return information regarding the project as it exists on GitHub.
 
     Else, return None.
     """
-    # project_urls can be None in the downloaded JSON.
-    project_url_dict = pypi_info.info.get("project_urls") or {}
-    for project_url in project_url_dict.values():
-        assert isinstance(project_url, str)
-        split_url = urllib.parse.urlsplit(project_url)
-        if split_url.netloc == "github.com" and not split_url.query and not split_url.fragment:
-            url_path = split_url.path.strip("/")
-            if len(Path(url_path).parts) == 2:
-                github_tags_info_url = f"https://api.github.com/repos/{url_path}/tags"
-                async with session.get(github_tags_info_url, headers=get_github_api_headers()) as response:
-                    if response.status == 200:
-                        tags: list[dict[str, Any]] = await response.json()
-                        assert isinstance(tags, list)
-                        return GithubInfo(repo_path=url_path, tags=tags)
+    if stub_info.upstream_repository is None:
+        return None
+    split_url = urllib.parse.urlsplit(stub_info.upstream_repository)
+    if split_url.netloc != "github.com":
+        return None
+    url_path = split_url.path.strip("/")
+    if len(Path(url_path).parts) != 2:
+        return None
+    github_tags_info_url = f"https://api.github.com/repos/{url_path}/tags"
+    async with session.get(github_tags_info_url, headers=get_github_api_headers()) as response:
+        if response.status == 200:
+            tags: list[dict[str, Any]] = await response.json()
+            assert isinstance(tags, list)
+            return GithubInfo(repo_path=url_path, tags=tags)
```

Comment:
You could do this with a smaller diff relative to the existing function:

```suggestion
    if stub_info.upstream_repository:
        split_url = urllib.parse.urlsplit(stub_info.upstream_repository)
        if split_url.netloc == "github.com" and not split_url.query and not split_url.fragment:
            url_path = split_url.path.strip("/")
            if len(Path(url_path).parts) == 2:
                github_tags_info_url = f"https://api.github.com/repos/{url_path}/tags"
                async with session.get(github_tags_info_url, headers=get_github_api_headers()) as response:
                    if response.status == 200:
                        tags: list[dict[str, Any]] = await response.json()
                        assert isinstance(tags, list)
                        return GithubInfo(repo_path=url_path, tags=tags)
```